src/script_lua.c

```diff
@@ -1039,14 +1039,62 @@ static void luaRemoveUnsupportedFunctions(lua_State *lua) {
     lua_setglobal(lua,"dofile");
 }
 
+/* This function installs metamethods in the global table _G that prevent
+ * the creation of globals accidentally.
+ *
+ * It should be the last to be called in the scripting engine initialization
+ * sequence, because it may interact with creation of globals.
+ *
+ * On Legacy Lua (eval) we need to check 'w ~= \"main\"' otherwise we will not be able
+ * to create the global 'function <sha> ()' variable. On Lua engine we do not use this trick
+ * so its not needed. */
```

Comment:
```suggestion
 * On Legacy Lua (eval) we need to check 'w ~= \"main\"' otherwise we will not be able
 * to create the global 'function <sha> ()' variable. On Functions Lua engine we do not use
 * this trick so it's not needed. */
```